sys/core/src/TxsCore.hs

```diff
@@ -172,16 +173,25 @@ import VarId
 -- | TorXakis core main api -- start
 runTxsCore :: Config -> StateT s IOC.IOC a -> s -> IO ()
 runTxsCore initConfig ctrl s0  =  do
-     _ <- runStateT (runTxsCtrl ctrl s0)
-               IOC.EnvC { IOC.config = initConfig
-                         , IOC.unid   = 0
-                         , IOC.params = initParams
-                         , IOC.state  = initState
-                         }
-     return ()
-       where initState = IOC.Noning
-             initParams =
-               Map.union ParamCore.initParams SolveDefs.Params.initParams
+      _ <- runStateT (runTxsCtrl ctrl s0)
+              IOC.EnvC { IOC.config = initConfig
+                        , IOC.unid   = 0
+                        , IOC.params = updateParams initParams
+                                        $ Config.configuredParameters initConfig
+                        , IOC.state  = initState
+                        }
+      return ()
+      where initState = IOC.Noning
+            initParams =
+              Map.union ParamCore.initParams SolveDefs.Params.initParams
+
```

Comment:
We need some documentation on what the `updateParams` function does.

Reply:
You can even put the types in multiple lines (making the line shorter than 80 characters):

```haskell
-- | This function does ...
updateParams :: ParamCore.Params -- ^ Foo ...
             -> [(Config.ParameterName,Config.ParameterValue)] -- ^ Bar ...
             -> ParamCore.Params
```

Reply:
Oh, and I miss a space after the comma :D

Reply:
Fixed and moved to Config